relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,242 @@
+use std::borrow::Cow;
+use std::sync::Arc;
+
+use relay_config::Config;
+use relay_dynamic_config::GlobalConfig;
+use relay_system::{Addr, AsyncResponse, FromMessage, Interface, Service};
+use reqwest::Method;
+use serde::{Deserialize, Serialize};
+use tokio::sync::{mpsc, watch};
+
+use crate::actors::upstream::{
+    RequestPriority, SendQuery, UpstreamQuery, UpstreamRelay, UpstreamRequestError,
+};
+use crate::utils::SleepHandle;
+
+/// The result of sending a global config query to upstream. It can fail both in sending it,
+/// and in the response.
+type UpstreamQueryResult =
+    Result<Result<GetGlobalConfigResponse, UpstreamRequestError>, relay_system::SendError>;
+
+/// The response of a fetch of a global config from upstream.
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfigResponse {
+    #[serde(default)]
+    global: Option<GlobalConfig>,
+}
+
+/// The request to fetch a global config from upstream.
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfig {
+    pub global: bool,
```

Comment:
Does this need to be public?

```suggestion
struct GetGlobalConfig {
    global: bool,
```

Reply:
+1, better not to have structs with mixed visibility. Either the entire fields are public (a data struct), or there are accessor methods.